### ISSUE TITLE
Issues training on new speaker

### ISSUE THREAD

**Issue by wanshun123** [keithito/tacotron#234]
I am trying to train a model to output a Trump voice by starting with the pretrained model from http://data.keithito.com/data/speech/tacotron-20180906.tar.gz (trained on LJSpeech to 441,000 steps) and then loading that model and training on data from https://github.com/aomv/voiceloop-in-the-wild-experiments/tree/master/data/donald-trump/data, which contains ~2 hours of speech divided into wav files 3 to 4 seconds in length. 

After preprocessing, training up to step 461,000 (ie. an extra 20k steps from the LJSpeech trained model), and leaving hparams.py untouched, I have the following results - some sentences are audible and complete, many aren't:

[results.zip](https://github.com/keithito/tacotron/files/2602387/results.zip)

And the training log:

[trainlog.zip](https://github.com/keithito/tacotron/files/2602392/trainlog.zip)

There is a steadily decreasing loss down to an average of ~0.097, and while the current results are abysmal, the results after 20k steps trained on the new data are noticeably better than the results after 5k or 10k steps. But as training this is a slow process - it takes around an hour per 1k steps on my Amazon EC2 p2.xlarge machine costing $0.90/hour - I am wondering if the results will still be poor after say 50k or 100k steps given how slowly the loss is decreasing.

In particular, I have concerns about custom data I'm using, for example: 

- in using such short audio files that were split automatically, there is often an incomplete word said at the start or end of the file, eg "-etely going to try to start commu-" which may be transcribed as "going to try to start", and the model will therefore pick up noise that doesn't represent a word. Should I be using audio files of say 10 seconds in length instead to minimise this?
- the length of audio recordings vs. character length of the transcription is inconsistent: [data_properties.zip](https://github.com/keithito/tacotron/files/2602425/data_properties.zip) (in my data most of the transcriptions are 30 to 40 characters in length, but there's little or no relationship between transcription length and length of the audio file).

What can I do to improve results, including making the data better or tweaking any parameters? Any advice appreciated.

**Comment by vaibhavthapliyal** [comment]
I think the pre trained model is trained upto 841000 steps not 441000 steps.

**Comment by isaiah658** [comment]
> What can I do to improve results, including making the data better or tweaking any parameters? Any advice appreciated.

I think you will get better results if you either find or make a better dataset. Having incomplete words at the start and end is most likely going to ruin everything. It's going to be better to cherry pick good sentences and remove sentences that have strange pauses, large changes in voice inflection, background noise, etc. From my experience, I was able to get an intelligible sounding voice with only 40 minutes of audio by training on top of the already trained LJ Speech model. After about 15k steps the audio is good, but it does have a tin robotic sound in the voice.

**Comment by wanshun123** [comment]
I've done a lot of testing with trying to train to a new speaker and found the model will indeed be useless if the data often contains partial words at the beginning or end. Only when using an extremely clean dataset where a new word is spoken right at the beginning of each audio file, and the audio file ends right when a word is completed could I generate reasonable results.
